.github/workflows/performance-tests.yml

```
@@ -0,0 +1,184 @@
+            python-version-short: '311'
+          - python-version: '3.12'
+            python-version-short: '312'
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)